FluentNest/Statistics.cs

```diff
@@ -139,29 +140,44 @@ public static class Statistics
         public static AggregationContainerDescriptor<T> DistinctBy<T>(this AggregationContainerDescriptor<T> agg, Expression<Func<T, object>> fieldGetter) where T : class
         {
             var aggName = fieldGetter.GetAggName(AggType.Distinct);
+            var namedField = Names.GetNameFromGetFieldNamed(fieldGetter.Body);
+
+            if (namedField != null)
+            {
+                return agg.Terms(aggName, x => x.Field(namedField).Size(int.MaxValue));
+            }
+
             return agg.Terms(aggName, x => x.Field(fieldGetter).Size(int.MaxValue));
         }
 
-        public static AggregationContainerDescriptor<T> TopHits<T>(this AggregationContainerDescriptor<T> agg, int size, params Expression<Func<T, object>>[] fieldGetter) where T : class
+        public static AggregationContainerDescriptor<T> TopHits<T>(this AggregationContainerDescriptor<T> agg, int size, params Expression<Func<T, object>>[] fieldGetters) where T : class
         {
             var aggName = AggType.TopHits.ToString();
-            return agg.TopHits(aggName, x => x.Size(size).Source(i=>i.Includes(f=>f.Fields(fieldGetter))));
+
+            var namedFields = fieldGetters.Select(x => Names.GetNameFromGetFieldNamed(x.Body)).Where(x => !string.IsNullOrEmpty(x)).ToArray();
+            var fieldGettersWithoutNamedFields = fieldGetters.Where(x => Names.GetNameFromGetFieldNamed(x.Body) == null).ToArray();
+
+            return agg.TopHits(aggName, x => x.Size(size).Source(i => i.Includes(f => f.Fields(fieldGettersWithoutNamedFields).Fields(namedFields))));

```

Comment:
Calling fields 2 times like that add ? nice I would have bet it replaced

Reply:
Yeah, I poked the API a bit until I found out you could do this.